src/components/Articles.jsx

```diff
@@ -1,24 +1,30 @@
 import { useState } from "react";
-import { Link } from "react-router-dom";
 import { getArticles } from "./utils/utils";
-import { useParams } from "react-router-dom";
+import { useParams, useSearchParams, Link } from "react-router-dom";
 import { useEffect } from "react";
-import { getArticlesByTopic } from "./utils/utils";
 import ArticleCards from "./ArticleCards";
 
 function Articles() {
 const { topic_slug } = useParams();
-const [articles, setArticles] = useState([])
-const [isLoading, setIsLoading] = useState(true)
+const [articles, setArticles] = useState([]);
+const [isLoading, setIsLoading] = useState(true);
+const [searchParams, setSearchParams] = useSearchParams();
+const sort_by = searchParams.get("sort_by") || "created_at";
+const order = searchParams.get("order") || "desc";
```

Comment:
Don't really need to set a default value here. If one of the queries is undefined, setSearchParams will handle it for you